doc/ui-guidelines.md

```diff
@@ -0,0 +1,252 @@
+  [{:keys [on-press on-long-press icon]}]
+  (let [[pressed? set-pressed] (rn/use-state false)
+        theme                  (theme/use-theme-value)
+        on-press-in            (rn/use-callback #(set-pressed true))
+        on-press-out           (rn/use-callback #(set-pressed nil))]
+    [rn/pressable
+     {:style               (style/main pressed? theme)
+      :on-press            on-press
+      :on-press-in         on-press-in
+      :on-press-out        on-press-out
+      :on-long-press       on-long-press}
+     [quo.icons/icon icon]]))
+```
+
+
+- We no longer need to create an anonymous function for rendering. This removes unnecessary confusion and the need for specific knowledge on how it works and why it was needed.
+- `rn/use-state` is used instead of `reagent/atom`
+- State values no longer need to be dereferenced; they are accessible as regular symbols. This eliminates a common bug where the "@" symbol was inadvertently omitted.
+- `theme/with-theme` wrapper is not needed anymore, `(theme/use-theme-value)` hook can be used directly in the components
+- `:f>` not needed anymore, all components are functional by default
+- `rn/use-callback` hook should be used for anon callback functions
+
+> [!IMPORTANT]
+> DO NOT USE anon functions directly in the props
+ 
+BAD
+```clojure
+(defn view
+  []
+  (let [[pressed? set-pressed] (rn/use-state false)]
+    [rn/pressable
+     {:style               (style/main pressed?)
+      :on-press-in         #(set-pressed true)
+      :on-press-out        #(set-pressed nil)}]))
+```
+
+GOOD:
+```clojure
+(defn view
+  []
+  (let [[pressed? set-pressed] (rn/use-state false)
+        on-press-in            (rn/use-callback #(set-pressed true))
+        on-press-out           (rn/use-callback #(set-pressed nil))]
+    [rn/pressable
+     {:style               (style/main pressed?)
+      :on-press-in         on-press-in
+      :on-press-out        on-press-out}]))
+```
+
+## Global state and subscriptions
+
+For global state management, we utilize Re-frame subscriptions. They can be likened to React state. To obtain the state, `(rf/sub [])` is employed, and to modify it, `(rf/dispatch [])` is utilized. However, they update components in a similar manner to React states.
+
+```clojure
+(defn view
+  [{:keys [selected-tab]}]
+  (let [collectible-list (rf/sub [:wallet/all-collectibles])
+        on-collectible-press (rn/use-callback
+                              (fn [{:keys [id]}]
+                               (rf/dispatch [:wallet/get-collectible-details id])))]
+    [rn/view {:style style/container}
+     (case selected-tab
+       :assets       [assets/view]
+       :collectibles [collectibles/view {:collectibles         collectible-list
+                                         :on-collectible-press on-collectible-press}])
+       [activity/view]]))
+```
+
+## Regular atoms
+
+In certain instances, components utilized regular atoms; however, they should now be used with `rn/use-ref-atom`
```

Comment:
Hi @flexsurfer, Please can you elaborate why we are doing this? What issues we are facing if we just use regular atoms in these cases.

Reply:
The primary difference now is that we no longer utilize the `reagent render` function, eliminating the need for an initialization wrapper. Consequently, we are left with a single function, eliminating any designated area for defining the atom. As a result, the atom, along with callback functions, will be defined each time. This necessitates the use of `use-ref-atom` and `use-callback` hooks. This approach aligns with React's methodology, offering familiarity to React developers. Direct usage of `ref` is impractical because it represents a JavaScript value, and modifying it would require `(set! ref.current value)`, which is suboptimal. Therefore, employing `use-ref-atom` presents a more idiomatic solution.